llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -120,6 +120,34 @@ bool AMDGPUInstructionSelector::constrainCopyLikeIntrin(MachineInstr &MI,
          RBI.constrainGenericRegister(Src.getReg(), *SrcRC, *MRI);
 }
 
+// Returns true if this is a copy from physical reg to VCC.
+// The parameter IsValid further indicates if the copy is valid.
+bool AMDGPUInstructionSelector::isCopyPhysicalToVCC(
```

Comment:
isCopyVCC. This should handle virtual -> virtual, physical -> virtual, and virtual -> physical cases